File: 3719-longest-balanced-subarray-i/3719-longest-balanced-subarray-i.py
class Solution:
    def longestBalanced(self, nums: List[int]) -> int:
        res = 0
        for i in range(len(nums) - 1):
            odds = set()
            evens = set()
            for j in range(i, len(nums)):
                if nums[j] % 2:
                    odds.add(nums[j])
                else:
                    evens.add(nums[j])
                if len(odds) == len(evens):
                    res = max(res, j - i + 1)
        return res

File: 3719-longest-balanced-subarray-i/3719-longest-balanced-subarray-i.java
class Solution {
    public int longestBalanced(int[] nums) {
        int ans = 0;
        Set<Integer> odds = new HashSet<>();
        Set<Integer> evens = new HashSet<>();
        for (int i = 0; i < nums.length; i++) {
            odds.clear();
            evens.clear();
            for (int j = i; j < nums.length; j++) {
                if (nums[j] % 2 == 0) odds.add(nums[j]);
                else evens.add(nums[j]);
                if (odds.size() == evens.size()) ans = Math.max(ans, j - i + 1);
            }
        }
        return ans;
    }
}